src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/rca/store/rca/cache/ShardRequestCacheRca.java

```diff
@@ -0,0 +1,223 @@
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.MetricFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.flow_units.ResourceFlowUnit;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotNodeSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.framework.api.summaries.HotResourceSummary;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.scheduler.FlowUnitOperationArgWrapper;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.reader.ClusterDetailsEventProcessor;
+import com.google.common.annotations.VisibleForTesting;
+import java.time.Clock;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.jooq.Record;
+import org.jooq.Result;
+
+/**
+ * Shard Request Cache RCA is to identify when the cache is unhealthy(thrashing) and otherwise, healthy.
+ * The dimension we are using for this analysis is cache eviction, hit count, cache current weight(size)
+ * and cache max weight(size) configured.
+ *
+ * <p>Cache eviction within Elasticsearch happens in following scenarios:
+ * <ol>
+ *   <li> Mutation to Cache (Entry Insertion/Promotion and Manual Invalidation)
+ *   <li> Explicit call to refresh()
+ * </ol>
+ *
+ * <p>The Cache Eviction requires that either the cache weight exceeds OR the entry TTL is expired.
+ * For Shard Request Cache, TTL is defined via `indices.requests.cache.expire` setting which is never used
+ * in production clusters and only provided for backward compatibility, thus we ignore time based evictions.
+ * The weight based evictions(removal from Cache Map and LRU linked List with entry updated to EVICTED) occur
+ * when the cache_weight exceeds the max_cache_weight, eviction.
+ *
+ * <p>The Entry Invalidation is performed manually on cache clear(), index close() and for cached results from
+ * timed-out requests. A scheduled runnable, running every 10 minutes cleans up all the invalidated entries which
+ * have not been read/written to since invalidation.
+ *
+ * <p>The Cache Hit and Eviction metric presence implies cache is undergoing frequent load and eviction or undergoing
+ * scheduled cleanup for entries which had timed-out during execution.
+ *
+ * <p>This RCA reads 'shardRequestCacheEvictions',  'shardRequestCacheHits', 'shardRequestCacheSize' and
+ * 'shardRequestCacheMaxSize' from upstream metrics and maintains collectors which keeps track of the time window
+ * period(tp) where we repeatedly see evictions and hits for the last tp duration. This RCA is marked as unhealthy
+ * if tp we find tp is above the threshold(300 seconds) and cache size exceeds the max cache size configured.
+ *
+ */
+public class ShardRequestCacheRca extends Rca<ResourceFlowUnit<HotNodeSummary>> {
```

Comment:
can we combine ShardRequestCacheRca and FieldDataCacheRca into one RCA and use different collector to collect those caches individually to reduce overhead ? Is there any benefit of creating two separate RCAs ?